tests/unit/systems/utils/tf.py

```diff
@@ -22,33 +22,44 @@
 
 
-def create_tf_model(cat_columns: list, cat_mh_columns: list, embed_tbl_shapes: dict):
-    inputs = {}  # tf.keras.Input placeholders for each feature to be used
-    emb_layers = []  # output of all embedding layers, which will be concatenated
-    for col in cat_columns:
-        inputs[col] = tf.keras.Input(name=col, dtype=tf.int64, shape=(1,))
-    # Note that we need two input tensors for multi-hot categorical features
-    for col in cat_mh_columns:
-        inputs[col] = (
-            tf.keras.Input(name=f"{col}__values", dtype=tf.int64, shape=(1,)),
-            tf.keras.Input(name=f"{col}__offsets", dtype=tf.int64, shape=(1,)),
-        )
-    for col in cat_columns + cat_mh_columns:
-        emb_layers.append(
-            tf.feature_column.embedding_column(
-                tf.feature_column.categorical_column_with_identity(
-                    col, embed_tbl_shapes[col][0]
-                ),  # Input dimension (vocab size)
-                embed_tbl_shapes[col][1],  # Embedding output dimension
+class ExampleModel(tf.keras.Model):
+    def __init__(self, cat_columns, cat_mh_columns, embed_tbl_shapes):
+        super(ExampleModel, self).__init__()
+
+        self.cat_columns = cat_columns
+        self.cat_mh_columns = cat_mh_columns
+        self.embed_tbl_shapes = embed_tbl_shapes
+
+        self.emb_layers = []  # output of all embedding layers, which will be concatenated
+        for col in self.cat_columns + self.cat_mh_columns:
+            self.emb_layers.append(
+                tf.feature_column.embedding_column(
+                    tf.feature_column.categorical_column_with_identity(
+                        col, self.embed_tbl_shapes[col][0]
+                    ),  # Input dimension (vocab size)
+                    self.embed_tbl_shapes[col][1],  # Embedding output dimension
+                )
             )
-        )
-    emb_layer = layers.DenseFeatures(emb_layers)
-    x_emb_output = emb_layer(inputs)
-    x = tf.keras.layers.Dense(128, activation="relu")(x_emb_output)
-    x = tf.keras.layers.Dense(128, activation="relu")(x)
-    x = tf.keras.layers.Dense(128, activation="relu")(x)
-    x = tf.keras.layers.Dense(1, activation="sigmoid", name="output")(x)
-
-    model = tf.keras.Model(inputs=inputs, outputs=x)
+        self.emb_layer = layers.DenseFeatures(self.emb_layers)
+        self.dense_layer = tf.keras.layers.Dense(128, activation="relu")
+        self.output_layer = tf.keras.layers.Dense(1, activation="sigmoid", name="output")
+
+    def call(self, inputs):
+        reshaped_inputs = {}
+        for input_name, input_ in inputs.items():
+            reshaped_inputs[input_name] = tf.reshape(input_, (-1, 1))
+
+        x_emb_output = self.emb_layer(reshaped_inputs)
+        x = self.dense_layer(x_emb_output)
+        x = self.output_layer(x)
+        return x
+
+
+def create_tf_model(cat_columns: list, cat_mh_columns: list, embed_tbl_shapes: dict):
```

Comment:
how much of this is a refactor vs fucntional change? It appears to be mostly a change to the input shape that's going on here. Which may be possible to make on the previous version with a change to the `shape` argument of the Keras Input

Reply:
I couldn't find any way to get this to work with the functional API and it reportedly won't work with the Sequential API either